script.js

```diff
@@ -0,0 +1,175 @@
+  );
+}
+
+function sleep() {
+  console.log(
+    `${this.name} which has ${this.age} years should go to sleep at 22:30. He/She did this in ${this.country}.`
+  );
+}
+function country() {
+  console.log(
+    `${this.name} is a ${this.sex} would like to travel to next countries ${this.travelList}.`
+  );
+}
+
+const person = new Person();
+
+acquaintance.apply(oleg);
+wakeUp.apply(oleg);
+sleep.apply(oleg);
+country.apply(oleg);
+
+acquaintance.call(iryna);
+wakeUp.call(iryna);
+sleep.call(iryna);
+country.call(iryna);
+
+/* Створити власну реалізацію методу .bind (підказка: в кінці треба помістити цю власну функцію у прототип -
+  Function.prototype.myOwnBind = function () { [тут_код_кастомного_bind] } */
+
+let myFunc = function (city) {
+  console.log(`My name is ${this.name} and I like my city ${city}.`);
+};
+
+Function.prototype.myOwnBind = function (oleg, ...args) {
+  let func = this;
+  return function () {
+    func.apply(oleg, [...args]);
+  };
+};
+
+let newFunc = myFunc.myOwnBind(oleg, `Kyiv`);
+newFunc();
+
+/* Cтворити функцію, котра приймає 2 параметри - об'єкти. Функція повинна перевіряти чи ці 2 обʼєкти абсолютно ідентичні та повертає результат у зрозумілому форматі */
+
+firstObj = { name: "Oleg", age: 38 };
+secondObj = { name: "Oleg", age: 38 };
+
+function objectIqualsOrNot(firstObj, secondObj) {
+  if (
+    firstObj === null ||
+    secondObj === null ||
+    firstObj === undefined ||
+    secondObj === undefined ||
+    firstObj === "" ||
+    secondObj === "" ||
+    firstObj === 0 ||
+    secondObj === 0
+  ) {
+    console.log(
+      `You entered a non-existing object. Please enter real objects for comparison.`
+    );
+  } else {
+    this.firstObj = firstObj;
+    this.secondObj = secondObj;
+    const result = Object.keys({ ...firstObj, ...secondObj }).every(
+      (key) => firstObj[key] === secondObj[key]
+    );
+    console.log(`These two objects iquals each other or not? It's ${result}.`);
+  }
+}
+objectIqualsOrNot();
+
+// objectIqualsOrNot(firstObj, secondObj);
+
+/* Створіть функцію-конструктор Calculator, який створює об’єкти з трьома методами:
+enterData - запитує два значення за допомогою prompt і запам’ятовує їх у властивостях об’єкта.
+calculateSum() повертає суму цих властивостей.
+calculateNSD() повертає результат пошуку НСД
+calculateNSK() повертає результат пошуку НСК */
+
+let firstNumber = +prompt(`Please enter first any number.`);
+let secondNumber = +prompt("Please enter second any number.");
+const enterData = { firstNumber, secondNumber };
+
+if (firstNumber <= 0 || firstNumber === "" || isNaN(firstNumber)) {
+  console.log(
+    `You entered incorrect first number. Please enter number more than 0.`
+  );
+} else if (secondNumber <= 0 || secondNumber === "" || isNaN(secondNumber)) {
+  console.log(
+    `You entered incorrect second number. Please enter number more than 0`
+  );
+} else {
+  function Calculator(enterData) {
```

Comment:
Better to call a function in `else` instead initializing it there
It should be something like
```suggestion
  function Calculator(numberOne, numberTwo) {
```